From 173beb6698538dcffefab36772e107ffb0b4fbbd Mon Sep 17 00:00:00 2001
From: Shaoqin Huang <shahuang@redhat.com>
Date: Mon, 28 Apr 2025 04:34:27 -0400
Subject: [PATCH 2/5] arm: Use arm_virt_compat_set() to apply the compat

RH-Author: Shaoqin Huang <shahuang@redhat.com>
RH-MergeRequest: 353: virtio-net: disable USO for virt-rhel9.6
RH-Jira: RHEL-80313
RH-Acked-by: Thomas Huth <thuth@redhat.com>
RH-Acked-by: Eric Auger <eric.auger@redhat.com>
RH-Commit: [2/2] 6e7a158e65296928040e70622b3cee59e45c1c36 (shahuang/qemu-kvm)

JIRA: https://issues.redhat.com/browse/RHEL-80313
Upstream Status: RHEL only

Since the pauth and uso both should apply for the latest machine type,
move them to the arm_virt_compat_set() which applies the compat to all
machine types automatically.

Signed-off-by: Shaoqin Huang <shahuang@redhat.com>
---
 hw/arm/virt.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/hw/arm/virt.c b/hw/arm/virt.c
index 896deaa025..2aef94e776 100644
--- a/hw/arm/virt.c
+++ b/hw/arm/virt.c
@@ -127,6 +127,10 @@ static void arm_virt_compat_set(MachineClass *mc)
                      arm_virt_compat_len);
     compat_props_add(mc->compat_props, arm_rhel_compat,
                      arm_rhel_compat_len);
+    compat_props_add(mc->compat_props, arm_rhel9_compat,
+                     arm_rhel9_compat_len);
+    compat_props_add(mc->compat_props, hw_compat_rhel_9,
+                     hw_compat_rhel_9_len);
 }
 
 #define DEFINE_VIRT_MACHINE_IMPL(latest, ...) \
@@ -3599,10 +3603,6 @@ DEFINE_VIRT_MACHINE(2, 6)
 
 static void virt_rhel_machine_9_6_0_options(MachineClass *mc)
 {
-    compat_props_add(mc->compat_props, arm_rhel9_compat, arm_rhel9_compat_len);
-
-    /* NB: remember to move this line to the *latest* RHEL 9 machine */
-    compat_props_add(mc->compat_props, hw_compat_rhel_9, hw_compat_rhel_9_len);
 }
 DEFINE_VIRT_MACHINE_AS_LATEST(9, 6, 0)
 
-- 
2.48.1

